### PR TITLE
Feature/reporting

### PR DESCRIPTION
**Pull request by SolomonShorser-OICR**
Adds some elastic queries and scripts that can be called from cron for automated reporting.

### CODE REVIEW

File: reporting/run_report.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Requirements:
+#  This script assumes that elasticsearch is running locally. It requires curl, mutt, and jq. mutt may require the gpgsm package.
+
+FROM_DATE=$1
+TO_DATE=$2
+QUERY_FILE=$3
+OUTPUT_FILE_PREFIX=$4
+HEADER=$5
+JQ_FILTER=$6
+MAIL_TO=$7
```
Comment (jweiser) [inline reporting/run_report.sh:25]: With 7 ordered arguments, would validation and/or named arguments be useful?

Comment (SolomonShorser-OICR) [reply reporting/run_report.sh]: This is intended to be run from a cron job, so it seemed like a lot of extra effort where it wouldn't be noticed.

Comment (SolomonShorser-OICR) [reply reporting/run_report.sh:25]: This is intended to be run from a cron job, so it seemed like a lot of extra effort where it wouldn't be noticed.

**Review by jweiser — CHANGES_REQUESTED** [review]
Overall looks good.  

I left one comment suggesting named arguments and validation for the bash script taking 7 values when executed.

Also, many values in the JSON values seem to be almost or exact duplicates.  Could they be consolidated so if they need to be modified, only making a change in one location is necessary?

*date_histogram
*aggs
*under "query.bool.must"
- query_string
*clientip values
*internal_IP:false
*probably_a_bot:false

- range

**Comment by SolomonShorser-OICR** [comment]
The JSON files are very similar because I wanted to keep them in a state that someone could execute them as a query by only substituting the start and end date of the ranges. I believe that refactoring them into smaller chunks that need to be stitched together in bash (ugh!) would add extra complications that aren't really necessary.

**Comment by jweiser** [comment]
> The JSON files are very similar because I wanted to keep them in a state that someone could execute them as a query by only substituting the start and end date of the ranges. I believe that refactoring them into smaller chunks that need to be stitched together in bash (ugh!) would add extra complications that aren't really necessary.

Okay, just a suggestion.  Would jq help with getting them put together if the duplications were localized to their own files?

**Comment by SolomonShorser-OICR** [comment]
> Okay, just a suggestion. Would jq help with getting them put together if the duplications were localized to their own files?

jq processes the output, so I'm not sure it would help there.

I really wanted to keep the query files as simple and stand-alone as possible. Currently, you could copy/paste these and run them as long as you add your own date-values, no extra work filling in substitutions or assembling various pieces is necessary. The only thing in them the _needs_ to be parameterized is the dates for the range. Everything else is not expected to change. ...Well, change _may_ happen (change _is_ the only Universal Constant, of course), but not as a regular part of reporting, it would most likely be due to major requirements change.

**Comment by jweiser** [comment]
> > Okay, just a suggestion. Would jq help with getting them put together if the duplications were localized to their own files?
> 
> jq processes the output, so I'm not sure it would help there.
> 
> I really wanted to keep the query files as simple and stand-alone as possible. Currently, you could copy/pate these and run them as long as you add your own date-values, no extra work filling in substitutions or assembling various pieces is necessary. The only thing in them the _needs_ to be parameterized is the dates for the range. Everything else is not expected to change. ...Well, change _may_ happen (change _is_ the only Universal Constant, of course), but not as a regular part of reporting, it would most likely be due to major requirements change.

Okay, that was the only point I wanted to discuss.  I'll approve the PR for merging.

**Review by jweiser — APPROVED** [review]

File: reporting/data_xfer_by_month_no_bots.query.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        "time_zone": "America/New_York",
+        "min_doc_count": 1,
+        "format": "yyyy-MM-dd"
+      },
+      "aggs": {
+        "Bytes_transferred": {
+          "sum": {
+            "field": "bytes",
+            "script": {
+              "source" : "((_value / 1024) / 1024) / 1024"
```
Comment (cookersjs) [inline reporting/data_xfer_by_month_no_bots.query.json:16]: I don't know if I'm getting too deep into the weeds by commenting on this, but why is it `/ 1024` three times?

**Review by cookersjs — APPROVED** [review]

File: reporting/data_xfer_by_month_no_bots.query.json (hunk 2 above)
Comment (SolomonShorser-OICR) [reply reporting/data_xfer_by_month_no_bots.query.json:16]: The data is saved as number of bytes. Dividing by 1024 gives KB, dividing again gives MB, 
dividing again: GB. We want the resulting format to be gigabytes.

I thought about simplifying it, but when changing precision of number of bytes, I find it easier to understand when written this way.

**Review by cookersjs — APPROVED** [review]

**Pull request merged**